Use _chsize_s on Windows for truncate

jaakkor2 opened the pull request:
Proposal to fix https://github.com/JuliaLang/julia/issues/24466 .  Should allow files >2 GB to be truncated on Windows.

jaakkor2 committed:
Use _chsize_s on Windows for truncate

Proposal to fix https://github.com/JuliaLang/julia/issues/24466 .  Should allow files >2 GB to be truncated on Windows.

vtjnash commented on src/support/ios.c:
```diff
@@ -566,7 +566,11 @@ int ios_trunc(ios_t *s, size_t size)
 #else
-        if (_chsize(s->fd, size) == 0)
+        #if !defined(__MINGW32__) || defined(MINGW_HAS_SECURE_API)
+            if (_chsize_s(s->fd, size) == 0)
```
I think we should just use this functoin everywhere. I don't see any reason we should need the alternate code path.

jaakkor2 committed:
Update ios.c

Checking Windows 32bit build https://julialang-s3.julialang.org/bin/winnt/x86/0.6/julia-0.6.1-win32.exe  , libgit2.dll does use `_chsize_c`, so both 32bit and 64bit should be able to use `_chsize_c`.

jaakkor2 commented:
Alternate code path removed.

vtjnash reviewed (approved):
Thanks!

(Although we might want to wait to merge until the windows buildbot is actually running, so please don't take a delay here the wrong way.)
